### ISSUE TITLE
Ordering by NULLS FIRST / LAST

### ISSUE THREAD

**Issue by gauda** [glebm/order_query#10]
Hallo and thanks for this neat gem!

I'm trying to implement an order with some custom sql but I do not get results. I have a field `posted_at` that sometimes may be `NULL`. The order I want to achieve is `posted_at DESC NULLS LAST`

I tried using order_query in different ways but to no avail. Can you please tell me if it's possible and point me in the right direction?

**Comment by glebm** [comment]
What database are you using?

**Comment by glebm** [comment]
As a quick hack for now while we figure out a proper solution, you could do this:

```ruby
order_query :my_order, 
   [:posted_at_or_zero, :desc, sql: "COALESCE(posted_at, '0000-00-00 00:00:00')"]

def posted_at_or_date_in_the_past
  posted_at || DateTime.new(0)
end
```

**Comment by gauda** [comment]
Looks good! That's postgresql so I had to change `order_query` to `COALESCE(posted_at, '1900-01-01 00:00:00')` and `posted_at_or_zero` to `posted_at || Time.utc(1900)` since postgresql does not allow zeros as date.

That's good enough for my application. Nevertheless it would be nice if you could do something like: `order_query :my_order, ['posted_at %order% NULLS LAST', :desc]` and order_query would replace `%order%` by `DESC` or `ASC`. This way all future problems, where the name of a field to order by is not enough, would be solved.

Thank you for your time and your hints how to solve this problem!

**Comment by bdtomlin** [comment]
I see that you did some work on this but it's well over a year old. I'm just curious what the status is and what your plans are for this feature.

**Comment by glebm** [comment]
The feature seems to work, but:

1. Tests are needed.
2. Branch needs rebasing.
3. Documentation needs updating.

I don't plan to do that myself, but if someone does all of the above and sends a PR I'll review it.

**Comment by bdtomlin** [comment]
Thanks for the reply. I looked at it and rebased from master with no issues. All the tests ran afterwards but the nulls feature still needs a some work and is not currently working.

Nulls can be especially tricky since many times you want to keep them in the result set but still have them sorted.

As I looked more into it I realized I would be better off implementing keyset queries in my own app. I think this will be the case for most people who need queries that this gem doesn't handle.

**Comment by glebm** [comment]
Status update: Added more tests to the branch and got some of them to pass.

**Comment by glebm** [comment]
Status update: The nulls-order branch now passes all the tests. Still needs testing with all the databases to be safe to merge (#15)

**Comment by glebm** [comment]
All done! Released in v0.4.0!
